halo2_proofs/src/plonk/mv_lookup.rs

```diff
@@ -0,0 +1,97 @@
+use super::circuit::Expression;
+use ff::Field;
+use std::fmt::{self, Debug};
+
+pub(crate) mod prover;
+pub(crate) mod verifier;
+
+/// Degree of lookup without inputs
+pub fn base_degree(table_degree: usize) -> usize {
+    // let lhs_degree = table_degree + inputs_expressions_degree + 1
+    // let degree = lhs_degree + 1
+    std::cmp::max(3, table_degree + 2)
+}
+
+pub fn degree_with_input(base_degree: usize, input_expression_degree: usize) -> usize {
+    base_degree + input_expression_degree
+}
+
+#[derive(Clone)]
+pub struct Argument<F: Field> {
```

Comment:
[critical-1]
In [[lookup.rs]](https://github.com/geometryresearch/halo2/blob/82f4d6da44358fb07123f022120f510bd33ebb9c/halo2_proofs/src/plonk/lookup.rs#L9) we have `pub name: &'static str,`, mv_lookup missing this field, resulting compatibility issue. 
Temporarily fixed in [mv lookup Argument name](https://github.com/scroll-tech/halo2/commit/e7eeb464275194e8af97fec4a6cdd2f0ca90b75d), needs double-check to make sure all usage is consistent with the original lookup